xmodule/library_content_block.py

```diff
@@ -527,16 +547,13 @@ def studio_post_duplicate(self, store, source_block):
 
         Otherwise we'll end up losing data on the next refresh.
         """
-        # The first task will be to refresh our copy of the library to generate the children.
-        # We must do this at the currently set version of the library block. Otherwise we may not have
-        # exactly the same children-- someone may be duplicating an out of date block, after all.
         user_id = self.get_user_id()
         user_perms = self.runtime.service(self, 'studio_user_permissions')
         if not self.tools:
             raise RuntimeError("Library tools unavailable, duplication will not be sane!")
-        self.tools.update_children(self, user_perms, version=self.source_library_version)
-
-        self._copy_overrides(store, user_id, source_block, self)
+        self.tools.trigger_duplicate_children(
+            user_id=user_id, user_perms=user_perms, source_block=source_block, dest_block=self
```

Comment:
extraneous user_id param, as user_id is acquired from the library tools object itself.

Reply:
👍🏻